src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -0,0 +1,119 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Fixture;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Utils;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Tests.Fixtures;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests.Controllers;
+
+public abstract class DataSetsControllerTests : IntegrationTestFixture
+{
+    private const string BaseUrl = "api/v1/data-sets";
+
+    public DataSetsControllerTests(TestApplicationFactory testApp) : base(testApp)
+    {
+    }
+
+    public class GetDataSetTests : DataSetsControllerTests
+    {
+        public GetDataSetTests(TestApplicationFactory testApp) : base(testApp)
+        {
+        }
+
+        [Theory]
+        [InlineData(DataSetStatus.Published)]
+        [InlineData(DataSetStatus.Deprecated)]
+        public async Task DataSetIsAvailable_Returns200(DataSetStatus dataSetStatus)
+        {
+            var dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatus(dataSetStatus)
+                .Generate();
```

Comment:
Don't have to do this, but you can actually use the generator's implicit conversion so we don't have to call `Generate` explicitly i.e.

```cs
DataSet dataSet = DataFixture
   .DefaultDataSet()
   .WithStatus(dataSetStatus);
```

Personally, I'm not particularly bothered either way, but it's something I've seen Ben doing a bit of so thought I'd mention it. Feel free to do whatever you'd prefer!

A similar thing could apply anytime we're calling `Generate`. Note this doesn't apply for `GenerateList` or `GenerateArray` variants where we should still use `var`.

Whatever we go with, we'll want to be consistent across all our tests we've written so far.